### PR TITLE
Add documentation for the Verifier contract

### PR DESCRIPTION
**Pull request by Cardosaum**

### CODE REVIEW

File: website/api/bonsai/verifier-contract/overview.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+The verifier contract incorporates the [`IRiscZeroVerifier` interface][IRiscZeroVerifier.sol], which defines the necessary data structures for the receipt and includes a `verify` method. This interface is implemented by the [`RiscZeroGroth16Verifier` contract][RiscZeroGroth16Verifier.sol], a stateless and immutable verifier designed for RISC Zero Groth16 SNARK proofs.
+
+Given the nature of the zkVM, any computation performed off-chain can be proven on-chain using a single verifier contract. This simplifies system architecture and eliminates the need for multiple contracts for different circuits. Even in the case of a system that needs to use custom circuits, it's possible to prove the resulting custom [receipts][term-receipt] within a [zkVM program][term-zkvm-program] and submit the resulting proof to the verifier contract.
+
+## Interface
+
+<!-- TODO: Link to auto-generated Solidity annotation docs -->
+
+TODO
```
Comment (Cardosaum) [inline website/api/bonsai/verifier-contract/overview.md:13]: Just not sure which content link here yet - I think it's in Victor's local branch?

Comment (nategraf) [reply website/api/bonsai/verifier-contract/overview.md]: I'll add this in a follow-up PR. I've removed this heading for now.

**Review by nategraf — APPROVED** [review]
I made some edits, and if you are happy with them, I'd say this doc is ready to land. As for sequencing, we will want to land this on top of Paul's reorg of the docs. If @pdg744 would prefer, he can go ahead and merge this into his own PR. Otherwise, we will wait and then land it after https://github.com/risc0/risc0/pull/1468/files is merged.

File: website/api/blockchain-integration/bonsai-on-eth.md [hunk 2]
```diff
@@ -21,29 +21,9 @@ Below is a diagram and overview of a typical application with RISC Zero on Ether
...
-
-[IRiscZeroVerifier.sol]: https://github.com/risc0/risc0-ethereum/blob/main/contracts/src/IRiscZeroVerifier.sol
-[RiscZeroGroth16Verifier.sol]: https://github.com/risc0/risc0-ethereum/blob/main/contracts/src/groth16/RiscZeroGroth16Verifier.sol
 [coprocessor-article]: https://www.risczero.com/news/a-guide-to-zk-coprocessors-for-scalability
-[etherscan-0x83C2e9CD64B2A16D3908E94C7654f3864212E2F8]: https://sepolia.etherscan.io/address/0x83c2e9cd64b2a16d3908e94c7654f3864212e2f8#code
 [foundry-template]: https://github.com/risc0/bonsai-foundry-template
-[groth16-article]: https://www.risczero.com/news/on-chain-verification
 [term-journal]: /terminology#journal
-[term-receipt]: /terminology#receipt
+[verifier-contract]: /api/blockchain-integration/contracts/verifier.md
```
Comment (flaub) [inline website/api/blockchain-integration/bonsai-on-eth.md:29]: Use relative paths
```suggestion
[verifier-contract]: ./contracts/verifier.md
```

File: website/api/blockchain-integration/contracts/verifier.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+We are working on process to provide opt-in upgradeability for the verifier contract, and we will have more information here in the future.
+
+## Addresses
+
+RISC Zero provides an already deployed verifier contract in the Sepolia network for your convenience.
+You can choose to use this contract or deploy your own.
+
+| Contract                      | Network   | Address                                                        |
+| ----------------------------- | --------- | -------------------------------------------------------------- |
+| [RiscZeroGroth16Verifier.sol] | [Sepolia] | [0x83C2e9CD64B2A16D3908E94C7654f3864212E2F8][sepolia-verifier] |
```
Comment (pdg744) [inline website/api/blockchain-integration/contracts/verifier.md:59]: Do we have a plan for versioning?

**Pull request merged**